### PR TITLE
Upgrade to .Net Core 3.0. 

### PR DESCRIPTION
**Pull request by cbehrends**
DbQuery has been deprecated so I've moved he impl over to DbSet.

### CODE REVIEW

**Review by MichalJankowskii — COMMENTED** [review]
I am not able to accept your changes because:
- added some custom unit tests
- removed support for DbQuery - it should be marked as Obsolete
- the main project should be targeted to netstandard 2.1

I created commit based on your changes with mentioned fixes already.

**Comment by cbehrends** [comment]
Glad the code was of use. 

I should warn you however that I've encountered some anomalies in relation to collections which forced me to stop using this library.  I was in a hurry at the time and had to get some code out, so I unfortunately did not write a test for it.  If I get some time I'll try to re-create it, if I can I'll write just the test and submit a PR.

File: src/Moq.EntityFrameworkCore.Examples/UsersServiceTest.cs [hunk 1]
```diff
@@ -12,6 +12,20 @@ public class UsersServiceTest
     {
         private static readonly Fixture Fixture = new Fixture();
 
+        [Fact]
```
Comment (MichalJankowskii) [inline src/Moq.EntityFrameworkCore.Examples/UsersServiceTest.cs:15]: What for is that test?

File: src/Moq.EntityFrameworkCore.Examples/Moq.EntityFrameworkCore.Examples.csproj [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
+    <TargetFramework>netcoreapp3.0</TargetFramework>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="AutoFixture" Version="4.8.0" />
-    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="2.2.1" />
-    <PackageReference Include="Moq" Version="4.10.1" />
+    <PackageReference Include="AutoFixture" Version="4.11.0" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="3.0.0" />
+    <PackageReference Include="Moq" Version="4.13.0" />
```
Comment (MichalJankowskii) [inline src/Moq.EntityFrameworkCore.Examples/Moq.EntityFrameworkCore.Examples.csproj:10]: Moq version update is not needed.

File: src/Moq.EntityFrameworkCore.Examples/Users/UsersContext.cs [hunk 3]
```diff
@@ -7,6 +7,6 @@ public class UsersContext : DbContext
     {
         public virtual DbSet<User> Users { get; set; }
 
-        public virtual DbQuery<Role> Roles { get; set; }
```
Comment (MichalJankowskii) [inline src/Moq.EntityFrameworkCore.Examples/Users/UsersContext.cs:10]: Deprecated - does not mean deletesd. Still, support should be in the package. Only Obsolete attribute should be added.

File: src/Moq.EntityFrameworkCore.Examples/Users/UsersService.cs [hunk 4]
```diff
@@ -1,4 +1,6 @@
-﻿namespace Moq.EntityFrameworkCore.Examples.Users
+﻿using System.Threading;
```
Comment (MichalJankowskii) [inline src/Moq.EntityFrameworkCore.Examples/Users/UsersService.cs:1]: Using should be in namespace

File: src/Moq.EntityFrameworkCore.Examples/Users/UsersService.cs [hunk 5]
```diff
@@ -15,6 +17,14 @@ public UsersService(UsersContext usersContext)
             this.usersContext = usersContext;
         }
 
+
+        public async Task<User> GetOne(int userId)
```
Comment (MichalJankowskii) [inline src/Moq.EntityFrameworkCore.Examples/Users/UsersService.cs:21]: What is the purpose of this method?

File: src/Moq.EntityFrameworkCore.Tests/Moq.EntityFrameworkCore.Tests.csproj [hunk 6]
```diff
@@ -1,12 +1,12 @@
...
   <PropertyGroup>
-    <TargetFramework>netcoreapp2.0</TargetFramework>
+    <TargetFramework>netcoreapp3.0</TargetFramework>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="2.2.1" />
-    <PackageReference Include="Moq" Version="4.10.1" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="3.0.0" />
+    <PackageReference Include="Moq" Version="4.13.0" />
```
Comment (MichalJankowskii) [inline src/Moq.EntityFrameworkCore.Tests/Moq.EntityFrameworkCore.Tests.csproj:9]: There is no need for new Moq package

File: src/Moq.EntityFrameworkCore.Tests/DbAsyncQueryProvider/InMemoryAsyncQueryProviderTests.cs [hunk 7]
```diff
@@ -76,7 +76,7 @@ public void Given_InMemoryAsyncQueryProvider_When_ExecutingQueryAsyncWithCancell
             this.inMemoryAsyncQueryProvider.ExecuteAsync(this.expression, CancellationToken.None);
 
             // Assert
-            this.queryProviderMock.Verify(x => x.Execute(this.expression));
+            this.queryProviderMock.Verify(x => x.Execute<object>(this.expression));
```
Comment (MichalJankowskii) [inline src/Moq.EntityFrameworkCore.Tests/DbAsyncQueryProvider/InMemoryAsyncQueryProviderTests.cs:79]: <object> is not needed

File: src/Moq.EntityFrameworkCore/DbAsyncQueryProvider/InMemoryAsyncEnumerable.cs [hunk 8]
```diff
@@ -1,4 +1,7 @@
-﻿namespace Moq.EntityFrameworkCore.DbAsyncQueryProvider
+﻿using System.Threading;
```
Comment (MichalJankowskii) [inline src/Moq.EntityFrameworkCore/DbAsyncQueryProvider/InMemoryAsyncEnumerable.cs:1]: Using in namespace

**Review by MichalJankowskii — CHANGES_REQUESTED** [review]
I added comments for the future.

Lib is already supporting EF3.0

**Pull request closed without merging**